algorithms/src/Kokkos_Sort.hpp

```diff
@@ -349,13 +349,13 @@ class BinSort {
 
  public:
   KOKKOS_INLINE_FUNCTION
-  void operator()(const bin_count_tag& /*tag*/, const int& i) const {
```

Comment:
I guess the changes to this file are not really necessary to fix the issue?
It's probably still a good idea to pass `const` primitive types by value consistently.

Reply:
essentially `i` got corrupted this was an attempt to fix it, and I thought I leave it this way.